plugin/src/main/groovy/com/novoda/staticanalysis/internal/detekt/DetektConfigurator.groovy

```diff
@@ -34,17 +38,17 @@ class DetektConfigurator implements Configurator {
                 return
             }
 
-            project.apply plugin: 'io.gitlab.arturbosch.detekt'
+            if (!project.plugins.hasPlugin(DETEKT_PLUGIN)) {
+                throw new GradleException(DETEKT_NOT_APPLIED)
+            }
 
             project.extensions.findByName('detekt').with {
                 // apply configuration closure to detekt extension
                 config.delegate = it
                 config()
             }
 
-            if (project.tasks.findByName('detektCheck')) {
-                configureToolTask()
-            }
+            configureToolTask()
```

Comment:
just looked at the implementation of `configureToolTask()`: does it mean that Detekt generates a single task per project? how do you setup separate source sets and filters on them?

Reply:
We rely on detekt to pick them up. In case the user doesn't provide any `input` detekt automatically looks up different modules.

If you want to have multiple source sets with different filters, you need to configure different `profiles`. As for now we can just access the `main` profile. @tasomaniac opened already an [issue](https://github.com/arturbosch/detekt/issues/705) on the detekt repo to expose other profiles as well.

Reply:
As we spoke IRL, @mr-archano is referring to our filter feature. I will add another subtask to evaluate whether we can support that.

Reply:
👍